packages/mock-account-provider/src/requesters.ts

```diff
@@ -104,39 +93,26 @@ export async function createPeer(
       }
     }
   }
-  return (await graphqlQuery({
-    resource: backendUrl,
-    method: 'post',
-    query: createPeerQuery,
-    variables: createPeerInput
-  })) as unknown as [PeerResponseElement]
-}
-
-export interface PeerResponseElement {
-  data: {
-    createPeer: {
-      code: string
-      success: boolean
-      message: string
-      peer: {
-        id: string
-        asset: {
-          code: string
-          scale: number
-        }
-        staticIlpAddress: string
+  return apolloClient
+    .mutate({
+      mutation: createPeerMutation,
+      variables: createPeerInput
+    })
+    .then(({ data }): CreatePeerMutationResponse => {
+      console.log(data)
+      if (!data.createPeer.success) {
```

Comment:
This was the one other place where I seemed to have to make a change--the `data` here has a `createPeer` object--I believe the "function name" in the gql--that constitutes another layer between the `data` and the `peer`. I don't think the structure of the response is inferred by the type system since it depends on the contents of the gql query. But the type system isn't fighting me as much any more, so still a big improvement.

Reply:
Yeah that looks right. I can look into getting better type inference for the response 👍